Common/include/toolboxes/ndflattener.hpp

```diff
@@ -37,8 +37,40 @@
 #include "../parallelization/mpi_structure.hpp"
 
 // --- Usage
-/*! \page ndflattener_usage Usage of NdFlattener
- * To demonstrate the usage of NdFlattener, let us collect information that depends on
+/*! \page ndflattener_usage NdFlattener usage
+ *
+ * # What is an NdFlattener?
+ *
+ * NdFlattener is a container to store a deep copy of multidimensional data in a single object that provides a
+ * generic []...[] interface. The data and their structure are stored in few contiguous arrays, allowing for an
+ * efficient MPI gathering operation.
+ *
+ * # Which problem does it solve?
+ * In SU2, there is a lot of multidimensional data: data depending on K>1 indices, like a wall property
```

Comment:
Thanks for added parts 👍 
A lot of work for the wall roughness feature that is not used (to my knowledge) but hopefully finds some good usage in the future 🔮 it certainly is one the better documented code bits :)